### ISSUE TITLE
Question about query generator using Bart & wizlnt

### ISSUE THREAD

**Issue by daje0601** [facebookresearch/ParlAI#4649]
> hi there - the query generator model for BB2 is just a regular BART model trained on the query generation task for Wizard of Internet (and multitasked with the MSC tasks to predict when to access memory); here's an example invocation for training a BART model with your own dataset:
> 
> ```shell
> python -m parlai.scripts.train_model -t custom_task --fp16 True --fp16-impl mem_efficient --gradient-clip 0.1 --log-every-n-secs 30 --lr-scheduler reduceonplateau --lr-scheduler-patience 3 --model-parallel True --model bart -o arch/bart_large --save-after-valid True --skip-generation True --optimizer adam -lr 7e-06 --warmup-updates 100 -vmm min -vmt ppl -vp 5 -veps 0.5 --update-freq 1 --batchsize 64 -tblog True --text-truncate 512 --truncate 512 --label-truncate 128 --model-file path_to_model
> ```
> 
> Where `-t custom_task` is your own dataset.
> 
> The BB2 model is similarly trained, just with `--model projects.blenderbot2.agents.blenderbot2:BlenderBot2FidAgent`. You can refer to the BB2 3B [model card](https://github.com/facebookresearch/ParlAI/blob/master/parlai/zoo/blenderbot2/model_card.md#hyperparameters) to see the hyperparameters for the 3B version; the 400m version will just need to include [BART's hyperparams instead](https://github.com/facebookresearch/ParlAI/blob/master/parlai/opt_presets/arch/bart_large.opt)

Hello,
First of all, I was very happy that I could understood and used parlai.

I'm trying to reproduct a query generator by looking at the code above it.
```
!python -m parlai.scripts.train_model -t wizard_of_internet \
 --fp16 True --fp16-impl mem_efficient --gradient-clip 0.1 \
 --log-every-n-secs 30 --lr-scheduler reduceonplateau --lr-scheduler-patience 3 \
 --model-parallel True --model bart \
 --activation gelu --attention-dropout 0.0 --dict-file zoo:bart/bart_large/model.dict \
 --dict-tokenizer gpt2 --dropout 0.1 --embedding-size 1024 --embeddings-scale False \
 --ffn-size 4096 --force-fp16-tokens True --fp16 True --init-model zoo:bart/bart_large/model \
 --learn-positional-embeddings True --model bart --n-decoder-layers 12 --n-encoder-layers 12 \
 --n-heads 16 --n-positions 1024 --variant bart \
 --save-after-valid True \
 --skip-generation True --optimizer adam -lr 7e-06 --warmup-updates 100 -vmm min \
 -vmt ppl -vp 5 -veps 0.5 --update-freq 1 --batchsize 8  --eval-batchsize 8 -tblog True \
 --text-truncate 512 --truncate 512 --label-truncate 128 --model-file content/model --save_every_n_secs 60  \
 --load-from-checkpoint True -wblog True --wandb-log True --wandb-name query_generator \
 --wandb-project BB2 
```
I wrote the code as above, and used the wizard of internet dataset.
Due to memory issues, batch_size has been changed from 64 -> 8 in Colab.

```
!parlai eval_model -mf content/model -t wizard_of_internet -dt valid -bs 8
```
The eval was conducted with the code above, and the results of loss(2.875) and ppl(17.73) were obtained.

However, the Issue is that it does not act as a query generator.
If you look at the picture below, you can see that the chatbot is having a conversation.
![image](https://user-images.githubusercontent.com/73736988/177292337-20aac685-bec9-4126-b6e4-af870b6689b8.png)

I'm not sure what else I need to modify to make it work like the query_generator on [this page](https://parl.ai/docs/zoo.html?highlight=query) and this code(`parlai interactive -mf zoo:blenderbot2/query_generator/model`)

Please help me in which direction to go.

**Comment by daje0601** [comment]
In relation to the above question, Do I need to train by adding `--search-server http://search.api.url \` code?

**Comment by mojtaba-komeili** [comment]
We are glad that you enjoy using ParlAI. I see you are `-t wizard_of_internet` as your task. The `wizard_of_internet` task is a whole family of teachers (see the teachers [here](https://github.com/facebookresearch/ParlAI/blob/main/parlai/tasks/wizard_of_internet/agents.py)). In ParlAI you may specify the specific teacher you want to use using`:`. For example, `wizard_of_internet:SearchQueryTeacher` is the search query teacher. If you don't use the second part, ParlAI will use the Default teacher in that task, which in this case is the dialogue response teacher. Here your teacher was trained on dialogue responses (and not search queries). That's why you see those responses.

**Comment by daje0601** [comment]
> We are glad that you enjoy using ParlAI. I see you are `-t wizard_of_internet` as your task. The `wizard_of_internet` task is a whole family of teachers (see the teachers [here](https://github.com/facebookresearch/ParlAI/blob/main/parlai/tasks/wizard_of_internet/agents.py)). In ParlAI you may specify the specific teacher you want to use using`:`. For example, `wizard_of_internet:SearchQueryTeacher` is the search query teacher. If you don't use the second part, ParlAI will use the Default teacher in that task, which in this case is the dialogue response teacher. Here your teacher was trained on dialogue responses (and not search queries). That's why you see those responses.

Oh!!!! Really, I try it right now. 
Is there any other teacher I need like this to reproduce Blenderbot??

**Comment by mojtaba-komeili** [comment]
For the dialogue model, you need to blend in all the teachers listed in the project page (hence called Blender). Also, if you are interested in memory generation, take a look at the MSC teachers.
